AndroidManifest.xml

```diff
@@ -6,7 +6,8 @@
     android:versionName="1.0" >
 
     <application android:label="TimeKeepReceiver">
-        <receiver android:name="TimeKeep">
+        <receiver android:name="TimeKeep"
+        android:exported="false">
```

Comment:
```suggestion
        <receiver android:name="TimeKeep" android:exported="false">
```

Or:

```suggestion
        <receiver android:name="TimeKeep"
                  android:exported="false">
```

Reply:
Yes, I tested it and it's working fine.
The 'exported' attribute is only used to allow ro deny other applications to explicitly call the activity/service. From Google documentation 

> the activity is accessible to any app, and is launchable by its exact class name.

Intents are always received correctly :)

Reply:
I find [the `<receiver>` documentation](https://developer.android.com/guide/topics/manifest/receiver-element) more clear on this:

> Whether the broadcast receiver can receive messages from **non-system sources** outside its application

(Emphasis mine)